src/main/java/org/reactome/addlinks/fileprocessors/MGIFileProcessor.java

```diff
@@ -0,0 +1,52 @@
+package org.reactome.addlinks.fileprocessors;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class MGIFileProcessor extends FileProcessor{
+
+    public MGIFileProcessor(String processorName)
+    {
+        super(processorName);
+    }
+
+    public MGIFileProcessor()
+    {
+        super(null);
+    }
+
+    @Override
+    public Map<String, List<String>> getIdMappingsFromFile()
+    {
+        Map<String, List<String>> mappings = new HashMap<>();
+        Path inputFile = Paths.get(this.pathToFile.toAbsolutePath().toString().replace(".gz", ""));
+        try
+        {
+            for (String line : Files.readAllLines(inputFile).stream().collect(Collectors.toList())) {
+                List<String> tabSplit = Arrays.asList(line.split("\t"));
+                if (tabSplit.size() > 6) {
+                  String mgiId = tabSplit.get(0).split(":")[1];
+                  List<String> uniprotIds = Arrays.asList(tabSplit.get(6).split(" "));
```

Comment:
It would be good to have a constant for the context of the number 6 here.  

The conditional check might also be better to check if the value you're extracting from the `tabSplit` list has content rather than the size of the list.  Something similar to this approach https://github.com/reactome/add-links/pull/134/files?w=1#r370308255 might be useful.

Reply:
As has been discussed within this PR, it turned out you need to use the `Array.size() > int` approach to avoid `ArrayOutOfBoundsException`s